test/test_asymetric_ECC.py

```diff
@@ -101,3 +103,269 @@ def test_sign_text(self):
+
+        self.session = self.pkcs11.openSession(
+            self.slot, PyKCS11.CKF_SERIAL_SESSION | PyKCS11.CKF_RW_SESSION
+        )
+        self.session.login("1234")
+
+        self.ec_base_tmpl = [
+            (PyKCS11.CKA_TOKEN, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_KEY_TYPE, PyKCS11.CKK_EC_EDWARDS),
+            (PyKCS11.CKA_LABEL, "test"),
+            (PyKCS11.CKA_ID, (0x33,)),
+        ]
+
+        self.ec_public_tmpl = self.ec_base_tmpl + [
+            (PyKCS11.CKA_CLASS, PyKCS11.CKO_PUBLIC_KEY),
+            (PyKCS11.CKA_PRIVATE, PyKCS11.CK_FALSE),
+            (PyKCS11.CKA_ENCRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_VERIFY, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_WRAP, PyKCS11.CK_TRUE),
+        ]
+
+        self.ec_priv_tmpl = self.ec_base_tmpl + [
+            (PyKCS11.CKA_CLASS, PyKCS11.CKO_PRIVATE_KEY),
+            (PyKCS11.CKA_SENSITIVE, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_DECRYPT, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_SIGN, PyKCS11.CK_TRUE),
+            (PyKCS11.CKA_UNWRAP, PyKCS11.CK_TRUE),
+        ]
+
+        # see https://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/os/pkcs11-curr-v3.0-os.html#_Toc30061191
+        self.schemes = {
+            # scheme     : ("curve", phFlag)
+            "Ed25519": ("ed25519", None),
+            "Ed25519ctx": ("ed25519", False),
+            "Ed25519ph": ("ed25519", True),
+            "Ed448": ("ed448", False),
+            "Ed448ph": ("ed448", True),
+        }
+
+    def tearDown(self):
+        self.session.logout()
+        self.pkcs11.closeAllSessions(self.slot)
+        self.pkcs11.unload()
+        del self.pkcs11
+
+    def _get_eddsa_mechanism(self, phf=None, ctx=None):
+        if phf is None:
+            return PyKCS11.EDDSA_Mechanism()
+        elif ctx is None:
+            return PyKCS11.EDDSA_Mechanism(phFlag=phf)
+        else:
+            return PyKCS11.EDDSA_Mechanism(phFlag=phf, contextData=ctx)
+
+    def _generate_edwards_key_pair(self, curve):
+        # Setup the domain parameters, unicode conversion needed
+        # for the curve string
+        ec_params = PrivateKeyAlgorithmId(curve).dump()
+        ec_public_tmpl = self.ec_public_tmpl + [(PyKCS11.CKA_EC_PARAMS, ec_params)]
+
+        (pubKey, privKey) = self.session.generateKeyPair(
+            ec_public_tmpl,
+            self.ec_priv_tmpl,
+            mecha=PyKCS11.Mechanism(PyKCS11.CKM_EC_EDWARDS_KEY_PAIR_GEN, None),
+        )
+        self.assertIsNotNone(pubKey)
+        self.assertIsNotNone(privKey)
+
+        # test CK_OBJECT_HANDLE.__repr__()
+        text = str(pubKey)
+        self.assertIsNotNone(text)
+        return (pubKey, privKey)
+
+    def test_sign_integer(self):
+        toSign = 1234567890
+        mecha = PyKCS11.EDDSA_Mechanism()
+        (pubKey, privKey) = self._generate_edwards_key_pair("ed25519")
+
+        # sign/verify
+        try:
+            self.session.sign(privKey, toSign, mecha)
+        except PyKCS11.PyKCS11Error as e:
```

Comment:
The code should **always** raise the exception?
Maybe it is better to use `self.assertRaises()`, no?
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises